compass/ocean/tests/soma/default/analysis.py

```diff
@@ -0,0 +1,90 @@
+import numpy as np
+import xarray as xr
+import matplotlib.pyplot as plt
+
+from compass.step import Step
+
+
+class Analysis(Step):
+    """
+    A step for verifying that temperature and salinity profiles for SOMA case
+    from particle interpolations are consistent with initialized profiles
+
+    Attributes
+    ----------
+    resolution : str
+        The resolution of the test case
+    """
+    def __init__(self, test_case, resolution):
+        """
+        Create the step
+
+        Parameters
+        ----------
+        test_case : compass.ocean.tests.soma.default.Default
+            The test case this step belongs to
+
+        resolution : str
+            The resolution of the test case
+        """
+        super().__init__(test_case=test_case, name='analysis')
+        self.resolution = resolution
+
+        self.add_input_file(
+            filename='particle_output.nc',
+            target='../forward/analysis_members/lagrPartTrack.0001-01-01_00.00.00.nc')
+
+        self.add_output_file('particle_temperature_comparison.png')
+        self.add_output_file('particle_salinity_comparison.png')
+
+    def run(self):
+        """
+        Run this step of the test case
+        """
+        plt.switch_backend('Agg')
+
+        ds = xr.open_mfdataset('particle_output.nc')
+
+        # obtain particle data
+        zLevelParticle = ds.zLevelParticle.values
+        particleTemperature = ds.particleTemperature.values
+        particleSalinity = ds.particleSalinity.values
+
+        section = self.config['soma']
+        alpha = section.getfloat('eos_linear_alpha')
+        drho = section.getfloat('density_difference')
+        t0 = section.getfloat('surface_temperature')
+        s0 = section.getfloat('surface_salinity')
+
+        # hard coded factors from namelist file
+        gamma = 1.0 / 1250.0  # psu /m
+        beta = 0.05
+        zt = 300  # m
+        hshelf = 100  # m
+        h0 = 2400  # m
```

Comment:
These would ideally not be hard-coded but instead would come from config options